### PR TITLE
Enable JIT compilation for AArch64, PowerPC, SystemZ, RISCV

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable JIT compilation for AArch64, PowerPC, SystemZ, RISCV.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Abandoned.

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 2b3ac0491b2a95c0072bf1eeec8ef588b38b00cf with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/38217/2b3ac0491b2a95c0072bf1eeec8ef588b38b00cf/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: contrib/llvm-project-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -58,18 +56,25 @@ set (REQUIRED_LLVM_LIBRARIES
     LLVMDemangle
 )
 
-# if (ARCH_AMD64)
+if (ARCH_AMD64)
```
Comment (alexey-milovidov) [inline contrib/llvm-project-cmake/CMakeLists.txt:59]: Should we add RISC-V, PPC, S390X here, or disable JIT for them?

Comment (kitaisreal) [reply contrib/llvm-project-cmake/CMakeLists.txt:59]: I will add all architectures here.

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by kitaisreal** [comment]
Performance tests results for AARCH64:
https://s3.amazonaws.com/clickhouse-test-reports/38217/c760a4c82f2b6ae98e1e94d62ec4588f8398c568/performance_comparison_aarch64_[1_4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/38217/c760a4c82f2b6ae98e1e94d62ec4588f8398c568/performance_comparison_aarch64_[2_4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/38217/c760a4c82f2b6ae98e1e94d62ec4588f8398c568/performance_comparison_aarch64_[3_4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/38217/c760a4c82f2b6ae98e1e94d62ec4588f8398c568/performance_comparison_aarch64_[4_4]/report.html

**Pull request merged**